api/v1/announcements/PKPAnnouncementController.php

```diff
@@ -209,27 +205,22 @@ public function add(Request $illuminateRequest): JsonResponse
             return response()->json($errors, Response::HTTP_BAD_REQUEST);
         }
 
-        $announcement = Repo::announcement()->newDataObject($params);
-
         try {
-            $announcementId = Repo::announcement()->add($announcement);
+            $announcement = Announcement::create($params);
         } catch (StoreTemporaryFileException $e) {
-            $announcementId = $e->dataObject->getId();
+            $announcementId = $e->getDataObjectId();
             if ($announcementId) {
-                $announcement = Repo::announcement()->get($announcementId);
-                Repo::announcement()->delete($announcement);
+                Announcement::destroy([$announcementId]);
             }
             return response()->json([
                 'image' => [__('api.400.errorUploadingImage')]
             ], Response::HTTP_BAD_REQUEST);
         }
 
-        $announcement = Repo::announcement()->get($announcementId);
-
         $sendEmail = (bool) filter_var($params['sendEmail'], FILTER_VALIDATE_BOOLEAN);
 
         if ($context) {
-            $this->notifyUsers($request, $context, $announcementId, $sendEmail);
+            $this->notifyUsers($request, $context, $announcement->getKey(), $sendEmail);
```

Comment:
OK, let's try. There is a possibility that it might interact with schema's id attribute, I don't see anything right now but `id` is a primary attribute and the logic I've put to the new schema setting `origin` defines Mutators and `composed` attributes. Meaning in `announcement.json`:
```json
"id": {
	"type": "integer",
	"origin": "primary",
	"readOnly": true,
	"apiSummary": true
},
```